### ISSUE TITLE
Bug Report: Cannot Find Dictionary Module

### ISSUE THREAD

**Issue by bdorney** [cms-gem-daq-project/gem-light-dqm#24]
## Brief summary of issue

Trying to call either the `unpacker` or the `dqm` outside of the directory:

```
$BUILD_HOME/gem-light-dqm/gemtreewriter/src/dic/
```

Results in an error:

```bash
$ dqm /data/bigdisk/GEM-Data-Taking/GE11_QC8/Cosmics/run000061_LatencyScan_CERNQC8_2018-07-19_chunk_0.raw.root 
--==DQM Main==--
Error in <TCling::RegisterModule>: cannot find dictionary module EventDict_rdict.pcm
```

```bash
$ unpacker run000061_LatencyScan_CERNQC8_2018-07-19_chunk_0.dat sdram
[GEMUnpacker]: ---> Main()
Error in <TCling::RegisterModule>: cannot find dictionary module EventDict_rdict.pcm
```

Some environment variable is not set correctly?

### Types of issue

- [X] Bug report (report an issue with the code)
- [ ] Feature request (request for change which adds functionality)

## Expected Behavior

Should not need to call from directory:

```
$BUILD_HOME/gem-light-dqm/gemtreewriter/src/dic/
```

## Current Behavior

Need to call from above directory

### Steps to Reproduce (for bugs)

According to lightDQM manual no instructions are needed to setup any env for catching this file

## Your Environment

* Version used: branch `release-v3`
* Shell used: `bash`

**Comment by mexanick** [comment]
did you sourced `settings.sh`?

**Comment by bdorney** [comment]
Coming back to this issue:

```bash
% source settings.sh 
settings.sh:source:2: no such file or directory: bin/thisroot.sh
~/scratch0/CMS_GEM/CMS_GEM_DAQ/gem-light-dqm
% unpacker $BUILD_HOME/tmpData/newDetAMC13Readout/temp_chunk_31.dat sdram
[GEMUnpacker]: ---> Main()
Error in <TCling::RegisterModule>: cannot find dictionary module EventDict_rdict.pcm
```

The error `settings.sh:source:2: no such file or directory: bin/thisroot.sh` seems to be coming from `$GEM_DQM_ROOT_BASE` not being defined.  But on the 904 machines is this even necessary now since root is a system install and is version `6.X.Y`

Again moving to the directory mentioned above and executing does not throw the `TCling` error:

```bash
cd $BUILD_HOME/gem-light-dqm/gemtreewriter/src/dic/
% unpacker $BUILD_HOME/tmpData/newDetAMC13Readout/temp_chunk_31.dat sdram
[GEMUnpacker]: ---> Main()
```

This appears to create the `.raw.root` file with a nonzero size:

```
% ll $BUILD_HOME/tmpData/newDetAMC13Readout/temp_chunk_31.raw.root
-rw-r--r--. 1 dorney zh 1.8M Aug 29 17:42 /afs/cern.ch/user/d/dorney/scratch0/CMS_GEM/CMS_GEM_DAQ/tmpData/newDetAMC13Readout/temp_chunk_31.raw.root
```

But trying to create the `.analyzed.root` file throws the following error:

```
% dqm $BUILD_HOME/tmpData/newDetAMC13Readout/temp_chunk_31.raw.root
...
...
Info in <TProofLite::SetQueryRunning>: starting query: 1
Info in <TProofQueryResult::SetRunning>: nwrks: 0
Info in <TUnixSystem::ACLiC>: creating shared library /afs/cern.ch/user/d/dorney/scratch0/CMS_GEM/CMS_GEM_DAQ/gem-light-dqm/dqm-root/src/common/gemTreeReader_cxx.so
In file included from input_line_12:9:
In file included from /afs/cern.ch/user/d/dorney/scratch0/CMS_GEM/CMS_GEM_DAQ/gem-light-dqm/dqm-root/src/common/gemTreeReader.cxx:26:
In file included from /afs/cern.ch/user/d/dorney/scratch0/CMS_GEM/CMS_GEM_DAQ/gem-light-dqm/dqm-root/src/common/AMC13_histogram.cxx:1:
In file included from /afs/cern.ch/user/d/dorney/scratch0/CMS_GEM/CMS_GEM_DAQ/gem-light-dqm/dqm-root/src/common/AMC_histogram.cxx:1:
In file included from /afs/cern.ch/user/d/dorney/scratch0/CMS_GEM/CMS_GEM_DAQ/gem-light-dqm/dqm-root/src/common/GEB_histogram.cxx:1:
/afs/cern.ch/user/d/dorney/scratch0/CMS_GEM/CMS_GEM_DAQ/gem-light-dqm/dqm-root/src/common/VFAT_histogram.cxx:78:15: error: no member named 'CRRCcheck' in 'VFATdata'
    if (vfat->CRRCcheck()) {
        ~~~~  ^
/afs/cern.ch/user/d/dorney/scratch0/CMS_GEM/CMS_GEM_DAQ/gem-light-dqm/dqm-root/src/common/VFAT_histogram.cxx:108:47: error: use of undeclared identifier 'b1010'
    else if (FiredChannels->GetEntries() > 64*b1010->GetEntries()) {
                                              ^
Error in <ACLiC>: Dictionary generation failed!
Error in <TSelector::GetSelector>: The file /afs/cern.ch/user/d/dorney/scratch0/CMS_GEM/CMS_GEM_DAQ/gem-light-dqm/dqm-root/src/common/gemTreeReader.cxx++g does not define a class named gemTreeReader.
Error in <TProofLite::CopyMacroToCache>: could not create a selector from /afs/cern.ch/user/d/dorney/scratch0/CMS_GEM/CMS_GEM_DAQ/gem-light-dqm/dqm-root/src/common/gemTreeReader.cxx++g
Info in <TProofQueryResult::RecordEnd>: output list cloned successfully!
Lite-0: all output objects have been merged
DQM Complete.
```

Output in the error message has been suppressed.

Running on version:

```bash
(py2.7) [dorney@gem904qc8daq]~/scratch0/CMS_GEM/CMS_GEM_DAQ/gem-light-dqm/gemtreewriter/src/dic% git show
commit ffd54665b83dc814e56d7337817ed2973aaa017d
Author: Mykhailo Dalchenko <mexanick@users.noreply.github.com>
Date:   Wed Jun 20 13:57:08 2018 +0200

    Create README.md

...
...
(py2.7) [dorney@gem904qc8daq]~/scratch0/CMS_GEM/CMS_GEM_DAQ/gem-light-dqm/gemtreewriter/src/dic% git branch -a
* dev_v3_cc7
```

Where `dev_v3_cc7` was set to track @mexanick's branch of the same name.

**Comment by mexanick** [comment]
resolved by #29
